compiler/injectdestructors.nim

```diff
@@ -1016,7 +1016,7 @@ proc p(n: PNode; c: var Con; s: var Scope; mode: ProcessMode; tmpFlags = {sfSing
 
 proc sameLocation*(a, b: PNode): bool =
   proc sameConstant(a, b: PNode): bool =
-    a.kind in nkLiterals and a.intVal == b.intVal
+    a.kind in nkLiterals and b.kind in nkLiterals and a.intVal == b.intVal
```

Comment:
This inner proc is misleading, `nkLiterals` includes string and float literals, the only reason we can access `intVal` here is because it's only called in the `nkBracketExpr` branch below. For readability's sake I would suggest renaming to `sameIndexConstant` or just inlining it.